RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,302 @@
+
+The Tari communication network consists of a number of different entities that need to communicate in a distributed and ad-hoc manner. 
+The primary entities that need to communicate are Validator Nodes (VN), [Base Node]s (BN), [Wallet]s (W) and Token Wallet (TW).
+Here are some examples of different communication tasks that need to be performed by these entities on the Tari Communication network:
+- Base Nodes on the Base Layer need to propagate completed blocks and transactions to other Base Nodes using Gossip protocol based broadcasting.
+- Validator Nodes need to efficiently discover other Validator Nodes in the process of forming Validator Node committees. 
+- Wallets need to communicate and negotiate with other Wallets to create transactions. They also need the ability to submit transactions to the [mempool] of Base Nodes.
+- Token Wallets need to communicate with Validator Node [committee]s and other Token Wallets to construct and send DAN instructions.
+
+Here is an overview communication matrix that show which source entities need to communicate with other destination entities on the Tari Communication network:
+
+| Source  \  Destination | Validator Node | Base Node | Wallet | Token Wallet |
+|---                     |---             |---        |---     |---           |
+| Validator Node         | Yes            | Yes       | Yes    | Yes          |
+| Base Node              | No             | Yes       | No     | No           |
+| Wallet                 | No             | Yes       | Yes    | No           |
+| Token Wallet           | Yes            | No        | Yes    | Yes          |
+
+#### Consensus Nodes and Consensus Clients
+
+To simplify the description of the Tari communication network, the different entities with similar behaviour were grouped into two groups: Consensus Nodes and Consensus Clients.
+Validator Nodes and Base Nodes are Consensus Nodes(CN).
+Wallets and Token Wallets are Communication Clients(CC).
+CNs form the core communication infrastructure of the Tari communication network and are responsible for maintaining the Tari communication network by receiving, forwarding and distributing joining requests, discovery requests, data messages and routing information.
+CCs are different from CNs in that they do not maintain the network and they are not responsible for propagating any joining requests, discovery requests, data messages and routing information.
+They do make use of the network to submit their own joining requests and perform discovery request of other specific CNs and CCs when they need to communicate with them.
+Once a CC has discovered the CC or CN they want to communicate with, they will establish a direct P2P channel with them.
+The Tari communication network is unaware of this direct P2P communication once discovery is completed.
+
+The different entity types were grouped into the different communication node types as follows:
```

Comment:
I think this table should be added to the MUST requirement category.

Reply:
Changed to "MUST"